src/NHibernate.Test/TestCase.cs

```diff
@@ -151,20 +151,27 @@ public void TearDown()
 			string badCleanupMessage = null;
 			try
 			{
-				OnTearDown();
-				var wereClosed = _sessionFactory.CheckSessionsWereClosed();
-				var wasCleaned = CheckDatabaseWasCleaned();
-				var wereConnectionsClosed = CheckConnectionsWereClosed();
-				fail = !wereClosed || !wasCleaned || !wereConnectionsClosed;
-
-				if (fail)
+				try
+				{
+					OnTearDown();
+				}
+				finally
```

Comment:
A cleanup failure would hide the tear down failure. But cleanup has far less risk of failures unless the server is down (case where hiding the teardown failure does not matter much). And trying to guard against cleanup failure for not hiding tear down failure causes the code to be a bit convoluted.